src/package-tree.ts

```diff
@@ -23,11 +31,45 @@ export interface PackageTree<T> {
   dependencies: Array<PackageTree<T>>;
 }
 
-function generatePackageTree(pjson: string): PackageTree<PointOfInterest[]> {
-  throw new Error('not implemented');
-  // compute result
-  //   let result: PackageTree;
-  //   return result;
+/**
+ * Takes in the root directory of a project and returns returns a PackageTree
+ */
+export async function generatePackageTree(
+    rootDir: string,
+    customReadFilep: ReadFileP): Promise<PackageTree<undefined>> {
+  customReadFilep = customReadFilep || readFilep;
+
+  try {
+    // Step 0: read in package.json and package-lock.json
+    const pjsonPath = path.join(rootDir, 'package.json');
+    const pjson = await customReadFilep(pjsonPath, 'utf8');
+    const packageJson = JSON.parse(pjson);
+
+    const pjsonLockPath = path.join(rootDir, 'package-lock.json');
+    const pjsonLock = await customReadFilep(pjsonLockPath, 'utf8');
+    const packageLockJson = JSON.parse(pjsonLock);
+
+    // Step 1: Get the top level dependencies from pjson
+    const dependencies = {
+      ...packageJson.dependencies,
+      ...packageJson.devDependencies
+    };
+
+    // Step 2: Look at package-lock.json to find dependendencies
+    const result =
+        await getPackageTreeFromDependencyList(dependencies, packageLockJson);
+
+    const treeHead: PackageTree<undefined> = {
+      name: packageJson.name,
+      version: packageJson.version,
+      data: undefined,
+      dependencies: result
+    };
+
+    return treeHead;
+  } catch (error) {
+    throw error;
```

Comment:
Do we necessarily need to re-throw here?

Reply:
No, I removed this